### PR TITLE
New configuration UI, small fixes

### PR DESCRIPTION
**Pull request by infeeeee**
Configurator rewritten from the ground up. 

I used this Inquirer implementation: [InquirerPy](https://github.com/kazhala/InquirerPy) [Docs](https://inquirerpy.readthedocs.io/en/latest/) It had more features than the other one mentioned in #32 

In `MenuPreset` I removed the `modify_value_callback` argument. The current usage of this tag was replaced by `question_type`, and InquirerPy has very powerful filters to sanitize the inputs: [Filter docs](https://inquirerpy.readthedocs.io/en/latest/pages/dynamic.html#filter)

In `MenuPreset`, unused configuration options are not saved, simply omitted from the json. As per #73 they are not needed, we read them from defaults on startup.

TODO, Ideas: (maybe in another PR)
- [x] Use the `filepath` InquirerPy prompt for path like configurations: [Docs](https://inquirerpy.readthedocs.io/en/latest/pages/prompts/filepath.html) ~~I think `modify_value_callback` from #82 could be replaced with this?~~ No, it's just a select list
- [x] Use number prompt for numbers: https://inquirerpy.readthedocs.io/en/latest/pages/prompts/number.html
- [x] Cancel current QuestionPreset with Escape key, InquirerPy has some interesting KeyboardInterrupt features

---

Unrelated to this, standardized some parts of the code:
- One `RunCommand()` basically a wrapper around `subprocess.run()`. There were a lot of very similar implementations around the codebase.
- OsD constants are much shorter now, easier to remember them. From `OperatingSystemDetection.OS_FIXED_VALUE_MACOS` to `OsD.MACOS`

Other files:
- `ClassManager.py`, `ConfiguratorIO.py`: linting
- `Power.py`: No separate Callback needed, they can decide it from the topic.

I tested this only on my main computer. Please check if it's working as expected on a Mac.

### CODE REVIEW

**Comment by richibrics** [comment]
Tested on macOS, I think it's working correctly.

The manage entities page is shows like in the picture, so if you don't scroll down the entire page you don't see the add entity button. What if we move it to the top ? And I'd add a "Active entities" label above the list of entities to express that those are the active entities. What do you think ?
image

**Comment by richibrics** [comment]
macOS again: I don't think if it is a bug of the library, but if I am in a page and I press Escape then nothing happens. Then when I press a following key on the keyboard, it comes back to the previous page

**Comment by richibrics** [comment]
Update: It comes back with escape If you wait 3 seconds after pressing it. Instead if you press instantly another key after escape there is no delay

**Comment by infeeeee** [comment]
Escape: I also found that it's slow, for me it's not that much, it's a bit annoying, should I remove it?


The manage entities page : That's intended, I mean that's the default settings of max_height, it uses maximum 70% of the screen: https://inquirerpy.readthedocs.io/en/latest/pages/height.html#max-height
I can take a look how it looks like at 100%

**Comment by richibrics** [comment]
Entities: okay but what about if we move the Add button to the top ? It would never disappear in this way.
However setting to 100% the max-height (if doesn't look bad) could be an option too, in fact with the white space on the bottom it could seem like there's nothing else under the shown entities.

File: IoTuring/Configurator/MenuPreset.py [hunk 1]
```diff
@@ -97,42 +126,74 @@ def AddEntry(self, name, key, default=None, mandatory=False, display_if_key_valu
...
-                    else:
-                        value = q_preset.default
+                    question_options["message"] = q_preset.question + ":"
 
-                # Set and sanitize the value:
-                q_preset.SetValue(value)
-                # Add to answered questions:
-                self.results.append(q_preset)
+                    if q_preset.default is not None:
+                        question_options["default"] = q_preset.default
```
Comment (infeeeee) [inline IoTuring/Configurator/MenuPreset.py:162]: default value for yes-no questions should be converted to boolean here, not in `IoTuring/Configurator/ConfiguratorObject.py`

**Comment by richibrics** [comment]
Very good work, I read your code and seems good. You really did a great job ! 🚀

**Comment by infeeeee** [comment]
Do not pull yet, I have to update the new Disk entity first

**Comment by richibrics** [comment]
Yes sure

**Comment by infeeeee** [comment]
I think it's ready, questions, I'm not sure about:

- The `< Go back` button can also move to the top, next to the `+Add new` button, What do you think is it better if they are on the two sides of the list? I would move it up in every menu.
- Should I remove the Escape button function in the menu? While using it actually I got used to the small delay.

**Comment by richibrics** [comment]
I think for consistency I'd put the Back below the Add. Do you like it ?
I'd also leave the Escape button because however it's a great feature and very useful

**Review by richibrics — APPROVED** [review]

**Comment by infeeeee** [comment]
Ok, I will move id up. I had an early version where it was the first in every menu

**Pull request merged**